cmd/scollector/collectors/mem_windows.go

```diff
@@ -23,21 +21,144 @@ func c_simple_mem_windows() (opentsdb.MultiDataPoint, error) {
 	}
 	var md opentsdb.MultiDataPoint
 	for _, v := range dst {
-		Add(&md, "win.mem.vm.total", v.TotalVirtualMemorySize*1024, nil, metadata.Gauge, metadata.Bytes, "Number, in bytes, of virtual memory.")
-		Add(&md, "win.mem.vm.free", v.FreeVirtualMemory*1024, nil, metadata.Gauge, metadata.Bytes, "Number, in bytes, of virtual memory currently unused and available.")
-		Add(&md, "win.mem.total", v.TotalVisibleMemorySize*1024, nil, metadata.Gauge, metadata.Bytes, "Total amount, in bytes, of physical memory available to the operating system.")
-		Add(&md, "win.mem.free", v.FreePhysicalMemory*1024, nil, metadata.Gauge, metadata.Bytes, "Number, in bytes, of physical memory currently unused and available.")
-		Add(&md, osMemTotal, v.TotalVisibleMemorySize*1024, nil, metadata.Gauge, metadata.Bytes, "Total amount, in bytes, of physical memory available to the operating system.")
+		Add(&md, "win.mem.vm.total", v.TotalVirtualMemorySize*1024, nil, metadata.Gauge, metadata.Bytes, descWinMemVirtual_Total)
+		Add(&md, "win.mem.vm.free", v.FreeVirtualMemory*1024, nil, metadata.Gauge, metadata.Bytes, descWinMemVirtual_Free)
+		Add(&md, "win.mem.total", v.TotalVisibleMemorySize*1024, nil, metadata.Gauge, metadata.Bytes, descWinMemVisible_Total)
+		Add(&md, "win.mem.free", v.FreePhysicalMemory*1024, nil, metadata.Gauge, metadata.Bytes, descWinMemVisible_Free)
+		Add(&md, osMemTotal, v.TotalVisibleMemorySize*1024, nil, metadata.Gauge, metadata.Bytes, osMemTotalDesc)
 		Add(&md, osMemFree, v.FreePhysicalMemory*1024, nil, metadata.Gauge, metadata.Bytes, osMemFreeDesc)
-		Add(&md, osMemUsed, v.TotalVisibleMemorySize*1024-v.FreePhysicalMemory*1024, nil, metadata.Gauge, metadata.Bytes, "")
+		Add(&md, osMemUsed, v.TotalVisibleMemorySize*1024-v.FreePhysicalMemory*1024, nil, metadata.Gauge, metadata.Bytes, osMemUsedDesc)
 		Add(&md, osMemPctFree, float64(v.FreePhysicalMemory)/float64(v.TotalVisibleMemorySize)*100, nil, metadata.Gauge, metadata.Pct, osMemPctFreeDesc)
 	}
 	return md, nil
 }
 
+const (
+	descWinMemVirtual_Total = "Number, in bytes, of virtual memory."
+	descWinMemVirtual_Free  = "Number, in bytes, of virtual memory currently unused and available."
+	descWinMemVisible_Total = "Total amount, in bytes, of physical memory available to the operating system."
+	descWinMemVisible_Free  = "Number, in bytes, of physical memory currently unused and available."
+)
+
 type Win32_OperatingSystem struct {
 	FreePhysicalMemory     uint64
 	FreeVirtualMemory      uint64
 	TotalVirtualMemorySize uint64
 	TotalVisibleMemorySize uint64
 }
+
+func c_windows_memory() (opentsdb.MultiDataPoint, error) {
+	var dst []Win32_PerfRawData_PerfOS_Memory
+	var q = wmi.CreateQuery(&dst, "")
```

Comment:
the "" is for a filter. We only use it if we want to exclude something (like name <> _total). The actual query is created by the wmi package by looking at the dst object.